### PR TITLE
TIG-2643 update c++ driver version to 3.6.0

### PR DESCRIPTION
**Pull request by guoyr**
Most of the work was in the toolchain, which was already pushed. Green build here: https://evergreen.mongodb.com/version/genny_toolchain_0db5d1544746c1570371f51109a0a312a7215b65

### CODE REVIEW

File: src/gennylib/src/PoolFactory.cpp [hunk 1]
```diff
@@ -187,7 +187,7 @@ std::string PoolFactory::makeUri() const {
 }
 
 mongocxx::options::pool PoolFactory::makeOptions() const {
-    mongocxx::options::ssl sslOptions;
+    mongocxx::options::tls sslOptions;
```
Comment (richardsamuels) [inline src/gennylib/src/PoolFactory.cpp:190]: It's probably best to rename this to `tlsOptions`

File: src/gennylib/test/PoolFactory_test.cpp [hunk 2]
```diff
@@ -219,7 +219,7 @@ TEST_CASE("PoolFactory behavior") {
 
         auto factoryOpts = factory.makeOptions();
 
-        auto sslOpts = *factoryOpts.client_opts().ssl_opts();
+        auto sslOpts = *factoryOpts.client_opts().tls_opts();
```
Comment (richardsamuels) [inline src/gennylib/test/PoolFactory_test.cpp:222]: `sslOpts` -> `tlsOpts`

**Review by richardsamuels — APPROVED** [review]
LGTM, aside from renaming references of SSL to TLS

**Comment by guoyr** [comment]
evergreen merge

**Comment by guoyr** [comment]
evergreen merge

**Comment by guoyr** [comment]
evergreen merge

**Comment by guoyr** [comment]
evergreen merge

**Comment by guoyr** [comment]
evergreen merge

**Comment by guoyr** [comment]
evergreen merge

**Comment by guoyr** [comment]
evergreen merge

**Pull request merged**